pkg/controller/topology.go

```diff
@@ -181,6 +181,12 @@ func GenerateAccessibilityRequirements(
 			// Distribute out one of the OR layers in allowedTopologies
 			requisiteTerms = flatten(allowedTopologies)
 		} else {
+			if strictTopology {
+				// Don't specify any topology requirements because neither the PVC nor
+				// the storage class have limitations.
+				return nil, nil
+			}
```

Comment:
@msau42: the change itself makes sense for local volume CSI drivers like PMEM-CSI.

Where I am undecided is whether this should be controlled via `strictTopology` or some new flag. The current definition of "strict topology" in the top-level README explicitly describes it as "ignored for immediate binding", so checking it here changes the semantic.

As user of the flag I am fine with that change; it actually would be simpler.

Reply:
hmmmm thinking about it closer, this can break drivers that support topology, so it's probably better to have a new flag. For example, for a typical cloud volume that supports zones, you may have a cluster that is deployed in only a subset of the zones. Drivers would depend on this aggregation to get the list of all valid zones.

Going back to the local volume case, how would a driver that has a central controller pick which node should be provisioned? Wouldn't that driver already need to keep some list of all nodes in the cluster to choose from?

Reply:
> For example, for a typical cloud volume that supports zones, you may have a cluster that is deployed in only a subset of the zones. Drivers would depend on this aggregation to get the list of all valid zones.

Do those use `--strict-topology`? I'm not sure. But you are right, as we can't know whether they do and changing the meaning of the `--strict-topology` would break them, we have to add another flag.

> Going back to the local volume case, how would a driver that has a central controller pick which node should be provisioned? Wouldn't that driver already need to keep some list of all nodes in the cluster to choose from?

A central controller needs a way to communicate with its nodes when it wants to provision a volume there. Just knowing what its nodes are is not enough.

In PMEM-CSI, we let nodes register themselves with the central controller, which also stores information about a port on which the node service can be reached to provision volumes. The list of nodes from external-provisioner in the CreateVolumeRequest topology alone is not enough and redundant.

Reply:
I decided to call this new flag `--topology-hints` because it felt like in this particular case, external-provisioner was providing additional information that wasn't strictly required, but may be useful. As always, naming is up for debate.. :stuck_out_tongue_closed_eyes:

Reply:
I don't think topology hints really says anything useful about what it does. What about something like "no-immediate-topology" to mean that if it's immediate binding, we don't pass any topology down and let the driver decide.

Reply:
That's better. I've updated the commit accordingly, please take another look.